frontend/src/app/Catalog/CatalogView/CatalogView.tsx

```diff
@@ -85,8 +100,106 @@ const CatalogView = (props: CatalogViewProps) => {
 			setOpen(true);
 		}
 	}, [coursePreview, data]);
+	
+	
+	useEffect(() => {
+		if (course !== null) {	
+			if (false && legacyId !== null) {		
+				dispatch(gradeReset())
+				dispatch(enrollReset())
+				dispatch(fetchEnrollContext());
+				dispatch(fetchGradeContext());
+				// let temp = semester.split(' ');
+				// dispatch(fetchGradeFromUrl(`/grades/0-${legacyId}-all-all`));
+				// axios.get(`/api/enrollment/sections/${legacyId}/`).then((res) => {
+				// 	dispatch(fetchEnrollFromUrl(`/enrollment/0-${legacyId}-${temp[0]}-${temp[1]}-${res.data[0].sections[0].section_id}`))
+				// 	});	
+			} else {
+				dispatch(gradeReset())
+				dispatch(enrollReset())
+				dispatch(fetchEnrollContext());
+				dispatch(fetchGradeContext());
+			}
+		}
+	}, [course?.courseNumber]);
+
+
+	const gradesData = useSelector((state: State) => state.grade?.gradesData ?? null);
+	
+	//may need in the future if graphs are rewritten
+	const enrollmentData = useSelector((state: State) => state.enrollment?.enrollmentData ?? null);
+	
+	const gradesGraphData = useSelector((state: State) => state.grade.graphData ?? null);
+	
+	//may need in the future if graphs are rewritten
+	const enrollGraphData = useSelector((state: State) => state.enrollment.graphData ?? null);
+	
+	const gradesSelectedCourses = useSelector((state: State) => state.grade.selectedCourses ?? null);
+	
+	const enrollSelectedCourses = useSelector((state: State) => state.enrollment.selectedCourses ?? null);
+	
+	//may need in the future if graphs are rewritten
+	const enrollContext = useSelector((state: State) => state.enrollment.context.courses ?? null);
+
+	useEffect(() => {
+		console.log(courseNumber)
+		if (course !== null) {
+			if (legacyId !== null) {	
+				setTab(0)	
+				dispatch(gradeReset())
+				dispatch(enrollReset())
+				dispatch(fetchEnrollContext());
+				dispatch(fetchGradeContext());
+				
+				if (legacyId !== null) {
+					let temp = semester.split(' ');
+					dispatch(fetchGradeFromUrl(`/grades/0-${legacyId}-all-all`));
+					axios.get(`/api/enrollment/sections/${legacyId}/`).then((res) => {
+						dispatch(fetchEnrollFromUrl(`/enrollment/0-${legacyId}-${temp[0]}-${temp[1]}-${res.data[0].sections[0].section_id}`))
+						});
+				}
+			} else {
+				setTab(0)	
+				dispatch(gradeReset())
+				dispatch(enrollReset())
+				dispatch(fetchEnrollContext());
+				dispatch(fetchGradeContext());
+
+				let legacyId = (store.getState() as any).enrollment?.context?.courses?.find(
+					(c: any) => c.abbreviation === abbreviation && c.course_number === courseNumber
+				)?.id ?? null
+				
+				if (legacyId !== null) {
+					let temp = semester.split(' ');
+					dispatch(fetchGradeFromUrl(`/grades/0-${legacyId}-all-all`));
+					axios.get(`/api/enrollment/sections/${legacyId}/`).then((res) => {
+						dispatch(fetchEnrollFromUrl(`/enrollment/0-${legacyId}-${temp[0]}-${temp[1]}-${res.data[0].sections[0].section_id}`))
+						});
+				}
+			}
+		}
+	}, [course?.courseNumber]);
```

Comment:
Currently this `effect` relies on a great deal more variables, yet `course.courseNumber` is the only specified dependencies. This leads to using stale data if `courseNumber` updates before other dependencies that this `effect` implicitly relies on.

A good tool to help with this is ESLint. UseEffect is a hard hook to use right and is super easy to misuse so no worries.

Reply:
downloaded es lint. Added a couple more dependencies so now all dispatches aside from resets are in one useeffect